### PR TITLE
Lmfdb#4587 squarefree av search

### PR DESCRIPTION
**Pull request by BarinderBanwait**
This PR addresses #4587 and adds an advanced search option for whether or not an abelian variety isogeny class over a finite field is (geometrically) squarefree. The functionality can be seen on [pink](https://pink.lmfdb.xyz).

Tests have been added for this. It was also observed that some other test methods had duplicate tests -- these have been removed.

### CODE REVIEW

File: lmfdb/abvar/fq/test_browse_page.py [hunk 1]
```diff
@@ -91,13 +91,32 @@ def test_simple_search(self):
...
+        # search for squarefree and geometrically squarefree
+        self.check_args("/Variety/Abelian/Fq/?geom_squarefree=YesAndGeom", "1.2.ac")
+        self.not_check_args("/Variety/Abelian/Fq/?geom_squarefree=YesAndGeom", "2.2.ad_f")
+
+        # search for squarefree but not geometrically squarefree
+        self.check_args("/Variety/Abelian/Fq/?geom_squarefree=YesNotGeom", "2.2.ad_f")
+
+        # search for non squarefree
+        self.check_args("/Variety/Abelian/Fq/?geom_squarefree=No", "2.2.ae_i")
+        self.not_check_args("/Variety/Abelian/Fq/?geom_squarefree=No", "1.3.ab")
```
Comment (edgarcosta) [inline lmfdb/abvar/fq/test_browse_page.py:116]: This test will fail, as 1.3.ab shows up as an isogeny factor

**Comment by edgarcosta** [comment]
Apart from the minor test failure, it looks good to me.
